util/crc32c.cc

```diff
@@ -315,9 +315,16 @@ static inline void Slow_CRC32(uint64_t* l, uint8_t const **p) {
 
 static inline void Fast_CRC32(uint64_t* l, uint8_t const **p) {
 #ifdef __SSE4_2__
+#ifdef __LP64__
   *l = _mm_crc32_u64(*l, LE_LOAD64(*p));
   *p += 8;
 #else
+  *l = _mm_crc32_u32(*l, LE_LOAD32(*p));
+  *p += 4;
+  *l = _mm_crc32_u32(*l, LE_LOAD32(*p));
```

Comment:
I don't think so.  If I understand your question correctly, you're wondering if I need to advance the ```l``` pointer for the next 32 bits.  The answer is no because despite ```_mm_crc32_u64``` taking a ````uint64_t*```` as its first argument, it only ever sets the lower 32 bits of that.

I did verify that this returns the same results in 64 bit and 32 bit.  Note, however, I don't have a big-endian machine to test on, so I can't verify that.

Also, I was able to run the crc32c_test which also passes.

However, it does result in a compiler warning that I somehow missed.  So please wait for an updated commit.